core/deployment/src/test/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReaderTest.java

```diff
@@ -0,0 +1,79 @@
+package io.quarkus.deployment.configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+
+class BuildTimeConfigurationReaderTest {
```

Comment:
My recommendation is to add the test here: https://github.com/quarkusio/quarkus/tree/4e11ab3c285ff7568c693d47a515122a3e7503d4/integration-tests/test-extension/extension

And use a `QuarkusUnitTest`.

Reply:
What was the issue?

Reply:
You can selectively create an inner class with the mapping local to the test class and then manually register it with:

```java
.addAsResource(new StringAsset("MAPPING_NAME"), "META-INF/quarkus-config-roots.list")
```

It should only affect the specific test and shouldn't leak to other tests.